oeps/oep-0019-bp-developer-documentation.rst

```diff
@@ -0,0 +1,293 @@
+     changes, the README.rst needs to be updated.
+
+   * The README may also serve as an index to the published docs of the component.
+
+4. **How-To Documentation** provides step-by-step instruction in enabling and using a feature.
+
+   * These will be stored in Github co-located with the relevant code.
+
+5. **API Documentation** provide information on interfaces for integrating with or enhancing the platform.
+   These should be automatically generated, as possible. For example:
+
+   * REST API interfaces will be automatically generated using Swagger_.
+   * The Open edX Learning Exchange (OLX) format should be automatically generated.
+   * Plugin interfaces (e.g., Proctoring integration) should be automatically generated using Sphinx_.
+
+.. _Clean Code: https://www.oreilly.com/library/view/clean-code/9780136083238/
+.. _OEP-1: https://open-edx-proposals.readthedocs.io/en/latest/oep-0001.html
+.. _open-edx-proposals: https://github.com/edx/open-edx-proposals
+.. _invaluable technique: https://www.thoughtworks.com/radar/techniques/lightweight-architecture-decision-records
+.. _local decisions: https://github.com/edx/credentials/tree/master/docs/decisions
+.. _Swagger: https://swagger.io/
+.. _Sphinx: http://www.sphinx-doc.org/en/master/
+
+Tools and Location
+~~~~~~~~~~~~~~~~~~
+
+The recommended tools and locations for developer documentation are described below, inlined with the
+requirements they meet.
+
+* **Discoverable**. Users must be able to find the documentation that is relevant to their needs. This 
+  calls for a high-level index and search mechanism for finding content.
+
+  * **Search**. For the time-being, we will make use of Google and Github's search functionality for
+    finding content since all developer documentation will be in Github.  Certain IDEs may also have good
+    tools for searching.
+
+  * **Index**. In the future, we will invest time in a better automated indexing tool to aggregate and
+    structure the documentation. For the time being, we will manually create and maintain documentation
+    indices at:
+    
+    * edx-developer-docs_ will include the landing page for developer documentation and be a container of
+      orphaned and cross-repo developer documentation.
+
+    * docs.edx.org_ will be the main landing page for *all* Open edX documentation, including a link to the
+      developer documentation index listed above.
+
+* **Maintainable**. Writers, editors, and reviewers must be able to create and modify documentation without
+  too much effort. Anyone, inside or outside of edX, should have the ability to contribute to documentation.
+
+  * **Co-located**. The closer the documentation lives with its corresponding code, the more likely it will be
+    kept up-to-date.
+
+    * Repo-specific documentation will live within a "docs" folder within its repo. This includes local
+      "decisions", "how-tos", and README.rst. Documentation should not live within the code itself (e.g., in
```

Comment:
This makes it sound like the top-level README.rst should be under `docs`, which doesn't seem right.

Reply:
True.  Updated.